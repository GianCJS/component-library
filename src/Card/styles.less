@card-padding: @space-normal;

.card {
    padding: @card-padding;
    box-sizing: border-box;
    position: relative;
    border-radius: @base-border-radius;

    &.without-padding {
        padding: 0;
    }

    &.with-large-padding {
        padding: @card-padding * 2;
    }

    &:not(.transparent) {
        background: @card-background;
        box-shadow: @card-shadow;
    }

}

/* Card header */

.card-header {
    display: flex;
    justify-content: space-between;
    align-items: baseline;
    margin-bottom: @card-padding * .75;
}

.card-title-action {
    margin-left: @space-small;
}
