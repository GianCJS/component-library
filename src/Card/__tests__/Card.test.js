import React from 'react';
import { shallow } from 'enzyme';
import Card from '../index';

describe('Card Unit Test', () => {
    let props;
    let shallowedCard;

    const getCard = () => {
        if (!shallowedCard) {
            shallowedCard = shallow(<Card {...props} />);
        }
        return shallowedCard;
    };

    beforeEach(() => {
        props = {
            children: 'Panel Card content',
        };
        shallowedCard = undefined;
    });

    test('Component should render without crashing', () => {
        expect(getCard()).toBeDefined();
    });

    test('Component should render correctly', () => {
        expect(getCard()).toMatchSnapshot();
    });
});
