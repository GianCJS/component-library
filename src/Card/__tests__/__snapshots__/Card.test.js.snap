// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Unit Test Component should render correctly 1`] = `
<div
  className="card 
         
        
        
        
    "
>
  Panel Card content
</div>
`;
